tests/unit/helpers/indexing-helper-test.php

```diff
@@ -49,95 +58,114 @@ public function setUp() {
 	 */
 	public function test_construct() {
 		$this->assertAttributeInstanceOf( Options_Helper::class, 'options_helper', $this->instance );
+		$this->assertAttributeInstanceOf( Date_Helper::class, 'date_helper', $this->instance );
 	}
 
 	/**
-	 * Tests setting the indexing reason.
+	 * Tests start.
 	 *
-	 * @covers ::set_reason
+	 * @covers ::start
+	 * @covers ::set_first_time
+	 * @covers ::set_started
 	 */
-	public function test_set_reason() {
-		$reason = 'permalinks_changed';
+	public function test_start() {
+		$this->options_helper
+			->expects( 'set' )
+			->once()
+			->with( 'indexing_first_time', false );
+
+		$start_time = 160934509;
+
+		$this->date_helper
+			->expects( 'current_time' )
+			->once()
+			->withNoArgs()
+			->andReturn( $start_time );
 
 		$this->options_helper
 			->expects( 'set' )
 			->once()
-			->with( 'indexing_reason', $reason );
+			->with( 'indexation_started', $start_time );
 
-		$this->instance->set_reason( $reason );
+		$this->instance->start();
 	}
 
 	/**
-	 * Tests getting the indexing reason.
+	 * Tests finish.
 	 *
-	 * @covers ::get_reason
+	 * @covers ::finish
+	 * @covers ::set_started
+	 * @covers ::set_reason
 	 */
-	public function test_get_reason() {
+	public function test_finish() {
 		$this->options_helper
-			->expects( 'get' )
+			->expects( 'set' )
+			->once()
+			->with( 'indexation_started', null );
+
+		$this->options_helper
+			->expects( 'set' )
 			->once()
 			->with( 'indexing_reason', '' );
 
-		$this->instance->get_reason();
+		$this->instance->finish();
 	}
 
 	/**
-	 * Tests setting the indexing start time.
+	 * Tests setting the indexing reason.
 	 *
-	 * @covers ::set_started
+	 * @covers ::set_reason
 	 */
-	public function test_set_started() {
-		$start_time = 160934509;
+	public function test_set_reason() {
+		$reason = 'permalinks_changed';
 
 		$this->options_helper
 			->expects( 'set' )
 			->once()
-			->with( 'indexation_started', $start_time );
+			->with( 'indexing_reason', $reason );
 
-		$this->instance->set_started( $start_time );
+		$this->instance->set_reason( $reason );
 	}
 
 	/**
-	 * Tests getting the indexing start time.
+	 * Tests getting the indexing reason.
 	 *
-	 * @covers ::get_started
+	 * @covers ::has_reason
 	 */
-	public function test_get_started() {
+	public function test_has_reason() {
 		$this->options_helper
 			->expects( 'get' )
 			->once()
-			->with( 'indexation_started' );
+			->with( 'indexing_reason', '' );
 
-		$this->instance->get_started();
+		$this->instance->has_reason();
```

Comment:
Since the `has_reason` method returns a value I would also assert that this value is correct.

Reply:
The same applies to the `test_get_started` test and the `test_is_initial_indexing` test below.